### PR TITLE
Allow setting batch sizes in admin settings

### PR DESCRIPTION
**Pull request by marcelklehr**
Should give people more flexibility.

See https://github.com/nextcloud/recognize/issues/466

image

### CODE REVIEW

File: lib/BackgroundJobs/ClassifyFacesJob.php [hunk 1]
```diff
@@ -43,6 +41,6 @@ protected function classify(array $files) : void {
 	 * @return int
 	 */
 	protected function getBatchSize() :int {
-		return $this->config->getAppValue('recognize', 'tensorflow.purejs', 'false') === 'false' ? self::BATCH_SIZE : self::BATCH_SIZE_PUREJS;
+		return intval($this->config->getAppValue('recognize', 'faces.batchSize', '100'));
```
Comment (juliushaertl) [inline lib/BackgroundJobs/ClassifyFacesJob.php:44]: Maybe keep the default value as a constant?

File: lib/BackgroundJobs/ClassifyImagenetJob.php [hunk 2]
```diff
@@ -12,8 +12,6 @@
 
 class ClassifyImagenetJob extends ClassifierJob {
 	public const MODEL_NAME = 'imagenet';
-	public const BATCH_SIZE = 100; // 100 images
-	public const BATCH_SIZE_PUREJS = 25; // 25 images
```
Comment (juliushaertl) [inline lib/BackgroundJobs/ClassifyImagenetJob.php:16]: Just to double check with the pure js variant, could the increase of the default cause follow up issues with performance or a higher crash rate?

**Review by juliushaertl — APPROVED** [review]
Looks good, just two questions inline ;)

File: lib/BackgroundJobs/ClassifyImagenetJob.php (hunk 2 above)
Comment (marcelklehr) [reply lib/BackgroundJobs/ClassifyImagenetJob.php:16]: Mmmh, yeah, that's a good point.

File: lib/BackgroundJobs/ClassifyFacesJob.php (hunk 1 above)
Comment (marcelklehr) [reply lib/BackgroundJobs/ClassifyFacesJob.php]: Moved to ClassifierJob

File: lib/BackgroundJobs/ClassifyImagenetJob.php (hunk 2 above)
Comment (marcelklehr) [reply lib/BackgroundJobs/ClassifyImagenetJob.php:16]: Moved to ClassifierJob and AdminSettings. Musicnn and Movinet may be a little overwhelmed by the slightly higher defaults, but I think it should be fine.

Comment (marcelklehr) [reply lib/BackgroundJobs/ClassifyImagenetJob.php:16]: Alternatively, I could create a Settings service that keeps track of the defaults, that would maybe be cleaner

**Pull request merged**